docs/docs/routing.md

```diff
@@ -29,10 +28,37 @@ The exception to this rule is any file named `index.js`. Files with this name ar
 
 Note that if no `index.js` file exists in a particular directory that root page does not exist, and attempts to navigate to it will land you on a [404 page](/docs/add-404-page/). For example, `yoursite.com/information/contact` may exist, but that does not guarantee `yoursite.com/information` exists.
 
+To create several pages from a data model, you can use the `createPagesFromData` macro with the collection builder route. For example, if you have a graphql model for products called `Product`. To create a page for each product, create a file inside `src/pages` at whatever route you would like (e.g., `/products/{name}.js` or `/products/{id}.js`). Inside of the file, the default export must use the gatsby `createPagesFromData` macro. Here is an example:
+
+```js:title=/src/pages/products{name}.js
+import { createPagesFromData, graphql } from "gatsby"
+
+function ProductComponent() {
+  /* ... */
+}
+
+export default createPagesFromData(ProductComponent, `Product`)
+
+export const ProductPageQuery = graphql`
+  query ($name: String) {
+    Product(name: { $eq: $name }) {
+      id
+      name
+    }
+  }
+`
+```
+
+See the documentation [Collection Page Builder]() for more detail.
+
+To create a client side only route, you can use `[]` as interpolation keys as a file path inside `src/pages` (e.g., `users/[id].js`). See the documentation for [client-only routes](/docs/client-only-routes-and-user-authentication) for more detail.
+
 ### Pages created with `createPage` action
 
 Another way to create pages is in your `gatsby-node.js` file using the `createPage` action, a JavaScript function. When pages are defined this way, the path is explicitly set. For example:
 
+_Note: This is an advanced feature and most use-case can be handled by creating pages in the `src/pages` directory_
```

Comment:
Is it possible to give an example of such a use case? Should we maybe mention that it used to be a standard way of creating pages but is no longer?